exp/views/user.py

```diff
@@ -147,20 +207,28 @@ class ResearcherCreateView(LoginRequiredMixin, generic.CreateView):
         'is_superuser',
         'password'
     )
+    def dispatch(self, request, *args, **kwargs):
+        if request.user.is_org_admin or request.user.is_superuser:
+            return super().dispatch(request, *args, **kwargs)
+        else:
+            raise PermissionDenied # HTTP 403
 
     def post(self, request, *args, **kwargs):
         # TODO put this on the view so that we can send the user an email once their user is saved
         # TODO alternatively send the password in a post_save signal under certain conditions
         self.user_password = User.objects.make_random_password(length=12)
         form = self.get_form()
         query_dict = form.data.copy()
-        # implicitly add them to their creator's organization
-        query_dict.update(is_active=True, is_superuser=False, is_staff=False, password=self.user_password, organization=self.request.user.organization)
+        query_dict.update(is_active=True, is_superuser=False, is_staff=False, password=self.user_password)
         form.data = query_dict
         if form.is_valid():
+            # implicitly add them to their creator's organization
+            form.instance.organization = self.request.user.organization
             return self.form_valid(form)
         else:
+            self.object = None
```

Comment:
This was for when there was an error in the researcher create form.

Adding to the code you already had here, if you used an email that had been used before then `self.form_invalid` would get called.  That method then called `self.get_context_data` which called  `self.object`.  This would fail because there was no object, I think because none had been created?